src/Features/CSharp/Portable/CodeRefactorings/ConvertIfToSwitch/CSharpConvertToSwitchCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,293 @@
+                currentIfStatement = elseBody as IfStatementSyntax;
+                if (currentIfStatement == null)
+                {
+                    break;
+                }
+            }
+
+            context.RegisterRefactoring(new MyCodeAction(c =>
+            {
+                IEnumerable<SyntaxNode> GetSwitchSectionBody(StatementSyntax node)
+                {
+                    if (!(node is BlockSyntax block && block.Statements.Count == 0))
+                    {
+                        yield return node;
+                    }
+
+                    if (semanticModel.AnalyzeControlFlow(node).EndPointIsReachable)
+                    {
+                        yield return SyntaxFactory.BreakStatement();
+                    }
+                }
+
+                CasePatternSwitchLabelSyntax GetSwitchLabel(Pattern pattern)
+                {
+                    CasePatternSwitchLabelSyntax CasePatternSwitchLabel(TypeSyntax type, VariableDesignationSyntax designation)
+                        => SyntaxFactory.CasePatternSwitchLabel(
+                            SyntaxFactory.DeclarationPattern(type, designation),
+                            SyntaxFactory.Token(SyntaxKind.ColonToken));
+
+                    switch (pattern)
+                    {
+                        case Pattern.ByValue p:
+                            return SyntaxFactory.CasePatternSwitchLabel(
+                                SyntaxFactory.ConstantPattern(p.Expression),
+                                SyntaxFactory.Token(SyntaxKind.ColonToken));
+
+                        case Pattern.ByType p:
+                            return CasePatternSwitchLabel(p.Type, SyntaxFactory.SingleVariableDesignation(p.Identifier));
+
+                        case Pattern.Discarded p:
+                            return CasePatternSwitchLabel(p.Type, SyntaxFactory.DiscardDesignation());
+
+                        case Pattern.Guarded p:
+                            return GetSwitchLabel(p.Pattern).WithWhenClause(SyntaxFactory.WhenClause(p.Guard));
+
+                        default:
+                            Debug.Assert(false);
+                            return null;
+                    }
+                }
+
+                SwitchSectionSyntax SwitchSection(SyntaxList<SwitchLabelSyntax> labels, StatementSyntax body)
+                    => SyntaxFactory.SwitchSection(labels, SyntaxFactory.List(GetSwitchSectionBody(body)));
+
+                var sections = sectionData.Select(data => SwitchSection(
+                    SyntaxFactory.List<SwitchLabelSyntax>(data.patterns.Select(GetSwitchLabel)), data.body));
+                var sectionList = SyntaxFactory.List(sections);
+                if (elseBody != null)
+                {
+                    sectionList = sectionList.Add(SwitchSection(
+                        SyntaxFactory.SingletonList<SwitchLabelSyntax>(SyntaxFactory.DefaultSwitchLabel()), elseBody));
+                }
```

Comment:
in general, you always need a blank line between a close curly and anything other than another close curly.